cip-TOKEN-STD/cip-meiersi-Canton-Network-Token-Standard.md

```diff
@@ -0,0 +1,567 @@
+Apps and wallets are free to choose any of the HTTP endpoints to query.
+The registry's Daml code must be written such that none of the nodes can influence the integrity
+of the Daml transactions built using the data from the node's HTTP response.
+Apps and wallets can thus get Byzantine-Fault-Tolerance by querying the
+off-ledger data from a random URL prefix of the decentralized registry and
+retrying in case the transaction submission fails.
+
+Apps and wallet clients can discover a (decentralized) registry's URL prefixes
+by [querying the metadata](#cns-entry-metadata) of the CNS entry for the registry's
+`admin` party with the key `splice.lfdecentralizedtrust.org/registryUrls`, and
+parsing it as a comma-separated list of URLs.
+
+TODO: once reviewed -- create issue to remove `RegistryAppInstall` and switch to this discovery mechanism
+
+#### Metadata
+
+The standard employs metadata records to enable transporting additional data as
+part of the standardized workflows. These metadata records are text-based
+key-value maps. Concretely, they are present on
+
+- choice arguments: for the caller to provide extra context information,
+  e.g., the reason for withdrawing an allocation
+- choice results: for the implementation to provide extra information about choice execution,
+  e.g., the fees charged
+- interface views: for the implementation to provide extra information about the contract,
+  e.g., an associated account-id
+- data records: to associate extra information throughout workflow with that data record
+  e.g., an off-ledger correlation id for the trade that is being settled
+
+App implementors, standard proposals, and standards are free to define metadata
+keys provided they prefix them with a DNS subdomain that uniquely identifies the
+organization defining the key, thereby avoiding clashes between keys from
+different organizations.
+
+This approach follows the design of
+[Kubernetes Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/)
+and enables new metadata keys to be introduced both in a top-down fashion by
+defining their meaning as part of CIPs, as well as in a bottom-up fashion by
+keys being defined ad-hoc and adopted more widely purely based on their
+usefulness.
+
+The metadata keys defined by this CIP are defined as part of
+[splice](https://github.com/hyperledger-labs/splice) and thus prefixed with
+`splice.lfdecentralizedtrust.org/`. Concretely, this CIP introduces the following two
+metadata keys:
+
+- `splice.lfdecentralizedtrust.org/registry-urls`: a comma-separated list of URLs of
+  the off-ledger APIs of a registry, used to [discover the off-ledger APIs of registries](#off-ledger-api-discovery-and-access)
+- `splice.lfdecentralizedtrust.org/lock-context`: used on `Holding` contracts to
+  provide a human-readable description of the context for the lock on a holding
+
+TODO: once reviewed - create issue to remove metadata keys related to holding fees
```

Comment:
why?

Reply:
fair point, let's omit it. My impression is also that people don't actually care all that much about accrued holding fees and much more about the effective ones they get charged on a transaction (e.g. our finance team). So we may wanna make more sure that we can display all the fees accurately on all transfers (which might mean going through the burn/mint API and putting it in the choice result in some custom metadata there). But we don't need to decide on that here.